cosmic-usage-ui/src/main/resources/public/app/main.js

```diff
@@ -72,6 +150,16 @@ const app = {
     generateReportButtonOnClick: function(event) {
         event.preventDefault();
 
+        this.generateSortableReport();
+    },
+
+    generateSortableReport: function() {
+        var currentSorting = this.getCurrentSorting()
+        if (currentSorting.sortBy === '' && currentSorting.sortOrder === '') {
```

Comment:
I'd move this logic to the `getCurrentSorting` method and I'd write `currentSorting.sortBy = this.DEFAULT_SORT_BY;` and `currentSorting.sortOrder = this.DEFAULT_SORT_ORDER;`